pkg/conn/baseconn.go

```diff
@@ -56,14 +64,31 @@ type BaseConn struct {
 	DBConn *sql.Conn
 
 	RetryStrategy retry.Strategy
+
+	// for reset
+	ParentDB *BaseDB
```

Comment:
can we mark it _private_ to avoid mis-modified?

Reply:
sure, [Done](https://github.com/pingcap/dm/pull/266/files#diff-e25c1def89789065c4904c1d5adc0867R69) and in order to mock baseConn in test, change baseConn's constructor to public [here](https://github.com/pingcap/dm/pull/266/files#diff-e25c1def89789065c4904c1d5adc0867R73)